pkg/reconciler/common/releases.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	mf "github.com/manifestival/manifestival"
+	"golang.org/x/mod/semver"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+const (
+	KoEnvKey = "KO_DATA_PATH"
+)
+
+var cache = map[string]mf.Manifest{}
+
+func TargetVersion(instance v1alpha1.KComponent) string {
+	switch {
+	case instance.GetSpec().GetVersion() != "":
+		return instance.GetSpec().GetVersion()
+	case instance.GetStatus().GetVersion() != "":
+		return instance.GetStatus().GetVersion()
+	default:
+		return latestRelease(instance)
+	}
+}
+
+func TargetManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
+	return fetch(manifestPath(TargetVersion(instance), instance))
+}
+
+func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
```

Comment:
Can you add comments to InstalledManifest and TargetManifest? I think they're similar enough that it might be confusing to some.

Reply:
Done